controllers/istiorevision/istiorevision_controller.go

```diff
@@ -532,24 +503,28 @@ func (r *Reconciler) mapNamespaceToReconcileRequest(ctx context.Context, ns clie
 	}
 
 	// Check if the namespace references an IstioRevision in its labels
-	revision := getReferencedRevisionFromNamespace(ns.GetLabels())
-	if revision != "" {
-		requests = append(requests, reconcile.Request{NamespacedName: types.NamespacedName{Name: revision}})
+	revisionName := revision.GetReferencedRevisionFromNamespace(ns.GetLabels())
+	if revisionName != "" {
+		requests = append(requests, reconcile.Request{NamespacedName: types.NamespacedName{Name: revisionName}})
 	}
 	return requests
 }
 
 func (r *Reconciler) mapPodToReconcileRequest(ctx context.Context, pod client.Object) []reconcile.Request {
-	// TODO: rewrite getReferencedRevisionFromPod to use lazy loading to avoid loading the namespace if the pod references a revision directly
-	ns := corev1.Namespace{}
-	err := r.Client.Get(ctx, types.NamespacedName{Name: pod.GetNamespace()}, &ns)
-	if err != nil {
-		return nil
+	revisionName := revision.GetInjectedRevisionFromPod(pod.GetAnnotations())
+	if revisionName == "" {
+		ns := corev1.Namespace{}
+		err := r.Client.Get(ctx, types.NamespacedName{Name: pod.GetNamespace()}, &ns)
+		if err != nil {
+			return nil
+		}
+		revisionName = revision.GetReferencedRevisionFromNamespace(ns.GetLabels())
+		if revisionName == "" {
+			revisionName = revision.GetReferencedRevisionFromPod(pod.GetLabels())
+		}
```

Comment:
Does the namespace label take precedence over the pod label? 

Also, if they refer to different revisions, do we need to reconcile both revisions?

Reply:
Apparently it does. At least according to the tests.

Reply:
> Does the namespace label take precedence over the pod label?
Counter-intuitively, yes. Luckily, the existing test cases in the IstioRevision controller covered this already.

> Also, if they refer to different revisions, do we need to reconcile both revisions?
Hmm. I guess technically yes, to make sure that InUse can be flipped to `false` in the case that you previusly had a pod label and are now adding a namespace label. Will make the change.

Reply:
Rewrote the func in the latest commit. PTAL

Reply:
Yeah, but if you're adding a namespace label, then `mapNamespaceToReconcile` is called, so it's probably not necessary to handle this in `mapPodToReconcileRequest`. And I guess we don't really need to handle cases where the pod label or annotation changes, since that means that someone is manually messing with them.

Reply:
oh, you're right. We probably don't need the change then. I can revert the commit